packages/js/data/src/product-attributes/types.ts

```diff
@@ -0,0 +1,44 @@
+/**
+ * External dependencies
+ */
+import { DispatchFromMap } from '@automattic/data-stores';
+
+/**
+ * Internal dependencies
+ */
+import { CrudActions, CrudSelectors } from '../crud/types';
+
+type ProductAttribute = {
+	id: number;
+	slug: string;
+	name: string;
+	type: string;
+	order_by: string;
+	has_archives: boolean;
+};
+
+type Query = {
+	context?: string;
+};
+
+type ReadOnlyProperties = 'id';
+
+type MutableProperties = Partial<
+	Omit< ProductAttribute, ReadOnlyProperties >
+>;
+
+type ProductAttributeActions = CrudActions<
+	'ProductAttribute',
+	ProductAttribute,
+	MutableProperties
+>;
+
+export type ProductAttributeSelectors = CrudSelectors<
+	'ProductAttribute',
+	'ProductAttributes',
+	ProductAttribute,
+	Query,
+	MutableProperties
+>;
+
+export type ActionDispatchers = DispatchFromMap< ProductAttributeActions >;
```

Comment:
I am getting this error related to this type and `DispatchFromMap`:
`Type 'DispatcherMap' is missing the following properties from type 'DispatchFromMap<ProductAttributeActions>': createProductAttribute, updateProductAttribute, deleteProductAttributets(2739)`

Reply:
This might just be the way I am using it, and a conflict with `useDispatch`, I as using it as such:
```
const actions: ProductAttributesActions = useDispatch(
		PRODUCT_ATTRIBUTES_STORE_NAME
	);
```

Reply:
Moving the type to the end with `as ProductAttributesActions`, does fix it.

Reply:
I think this might the correct behavior since the type should be on the dispatch and not the actions themselves.